### PR TITLE
implement 0x17 - OR opcode #12

### PR DESCRIPTION
**Pull request by Noeljarillo**
Added OR opcode

Enhancement OpCode

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by enitrat** [comment]
Hey Noel, nice to see you here taking some initiatives! 

Before opening PRs, could you make sure that there is an existing issue and that you can ask to be assigned on it? This makes the open contributions easier to manage :)

If the issue doesn't exist yet, you can go to https://evm.codes, and copy-paste the corresponding opcode to the issue (you can copy the HTML code of the block to get a nice formatting)

After that, when you open a PR, you need to link the issue with the syntax `Closes #issue-number`. Thanks!

**Review by enitrat — COMMENTED** [review]
Can you add a testcase with values more complex than 0 and 1 ? 

e.g from the [legacy implementation](https://github.com/kkrt-labs/kakarot/blob/44dabfcab05a87f85bcfe0a457809059d11d83fb/tests/src/kakarot/instructions/test_comparison_operations.cairo#L391): `0x89 | 0xC5 = 0xCD`

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 1]
```diff
@@ -1654,6 +1654,39 @@ fn assert_sar(a: u256, b: u256, expected: u256) {
...
+    let mut ctx = setup_execution_context();
+    ctx.stack.push(0x01).unwrap();
+    ctx.stack.push(0x00).unwrap();
+
+    // When
+    ctx.exec_or();
+
+    // Then
+    assert(ctx.stack.len() == 1, 'stack should have one element');
+    assert(ctx.stack.peek().unwrap() == 0x01, 'stack top should be true');
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:1670]:
```suggestion
    assert(ctx.stack.peek().unwrap() == 0x01, 'stack top should 0x01');
```

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 2]
```diff
@@ -1654,6 +1654,39 @@ fn assert_sar(a: u256, b: u256, expected: u256) {
...
+    let mut ctx = setup_execution_context();
+    ctx.stack.push(0x00).unwrap();
+    ctx.stack.push(0x00).unwrap();
+
+    // When
+    ctx.exec_or();
+
+    // Then
+    assert(ctx.stack.len() == 1, 'stack should have one element');
+    assert(ctx.stack.peek().unwrap() == 0x00, 'stack top should be false');
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:1686]:
```suggestion
    assert(ctx.stack.peek().unwrap() == 0x00, 'stack top should be 0x00');
```

**Comment by Eikix** [comment]
Important update: we are gathering some bugs in the [Kakarot v0 codebase](https://github.com/kkrt-labs/kakarot), we need to make sure each issue and each PR in Kakarot-ssj is aware of the lists of known bugs. Look at this link everytime you take an issue and check your issue isn't targeted by a known bug.

**Comment by Eikix** [comment]
Important update: we are gathering some bugs in the [Kakarot v0 codebase](https://github.com/kkrt-labs/kakarot), we need to make sure each issue and each PR in Kakarot-ssj is aware of the lists of known bugs. Look at this link everytime you take an issue and check your issue isn't targeted by a known bug.

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 3]
```diff
@@ -1654,6 +1654,54 @@ fn assert_sar(a: u256, b: u256, expected: u256) {
     assert(ctx.stack.peek().unwrap() == expected, 'sar failed');
 }
 
+#[test]
+#[available_gas(20000000)]
+fn test__exec_or__should_pop_0_and_1_and_push_0xCD__when_0_is_0x89_and_1_is_0xC5() {
```
Comment (danilowhk) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:1659]: Didn't get this test, shouldn't exec_or return only 0 or 1?

Comment (enitrat) [reply crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:1659]: it's bitwise OR

Comment (Noeljarillo) [reply crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:1659]: Same tests as old version.
0x89 | 0xC5 = 0xCD
as is bitwise OR

**Pull request merged**